samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -918,14 +919,38 @@ def _generate_makefile_rule_for_lambda_resource(
     """
     target = _get_makefile_build_target(logical_id)
     resource_address = sam_metadata_resource.resource.get("address", "")
+    override_filename = f"zz_sam_cli_backend_{uuid.uuid4()}_override.tf"
+    backend_filename = f"{uuid.uuid4()}.tfstate"
+    init_backend_recipe = _generate_and_format_init_local_backend_recipe(override_filename, backend_filename)
     apply_command_template = "terraform apply -target {resource_address} -replace {resource_address} -auto-approve"
     apply_command_recipe = _format_makefile_recipe(apply_command_template.format(resource_address=resource_address))
     show_command_recipe = _format_makefile_recipe(
         _build_show_command(
             python_command_name, output_dir, resource_address, sam_metadata_resource, terraform_application_dir
         )
     )
-    return f"{target}{apply_command_recipe}{show_command_recipe}"
+    rm_override_file_recipe = _format_makefile_recipe(f"rm {override_filename}")
+    rm_backend_file_recipe = _format_makefile_recipe(f"rm {backend_filename}")
+    return (
+        f"{target}"
+        f"{init_backend_recipe}"
+        f"{apply_command_recipe}"
+        f"{show_command_recipe}"
+        f"{rm_override_file_recipe}"
+        f"{rm_backend_file_recipe}"
+    )
+
+
+def _generate_and_format_init_local_backend_recipe(override_filename: str, backend_filename: str) -> str:
+    command_list = [
+        f'echo "terraform {{" > {override_filename}',
```

Comment:
do you think this way will work in windows ?

Reply:
Right, this might not work. Maybe it's better to create another python script to fulfill this purpose. Let's discuss about what the best approach is. And then I will update it.

Reply:
After discussion, this PR only focuses on making it work for linux and mac. We will have another task to make it work for Windows